app/jobs/subscription_confirm_job.rb

```diff
@@ -62,7 +62,7 @@ def process_payment!(order)
     return unless order.payment_required?
 
     prepare_for_payment!(order)
-    order.process_payments!
+    order.process_payments!(true)
```

Comment:
I totally agree and in fact I'm one step ahead of you ;-) https://github.com/openfoodfoundation/openfoodnetwork/pull/6241/commits/df40352389db6285e4cb12e64d78321c8594dae3

I'm not sure why that commit didn't show up for you when you reviewed? Another reason to simplify/squash commits.

In any case, yes, the flag argument, even if it's named, isn't great. I considered a separate method but it got a little tangled with the other methods. I'll have another try at it soon.

Reply:
Here's a first attempt at removing the flag argument: 59e978eeda1995194c970bba154d6faaf2b29b9a. I was able to refactor some of the duplication but I didn't see a great way to get rid of it in the `Order` class.... Still 🤔

Reply:
Okay, I'm now more convinced that the separate method is the way to go, even if there is some duplication. The `process_payments!` method has logic at the end specific to checkout, which the `process_payments_offline!` method should not have 👍. I updated the linked hash.

Reply:
I usually review commit by commit. So I wrote the comment before seeing your later commit. And when I saw that, I could have deleted my comment but even though it was much better with a named parameter, I was still wondering if it could be better with a separate method.